// @quillforms/blocklib-statement-block/build-module/admin/controls.js
import { createElement } from "react";
/**
 * QuillForms Dependencies
 */
import { BaseControl, ControlWrapper, ControlLabel, ToggleControl, TextControl } from '@quillforms/admin-components';

/**
 * WordPress Dependencies
 */
import { Fragment } from 'react';
const statementControls = props => {
  const {
    attributes,
    setAttributes
  } = props;
  const {
    buttonText,
    quotationMarks
  } = attributes;
  return createElement(Fragment, null, createElement(BaseControl, null, createElement(ControlWrapper, {
    orientation: "horizontal"
  }, createElement(ControlLabel, {
    label: "Quotation marks"
  }), createElement(ToggleControl, {
    checked: quotationMarks,
    onChange: () => setAttributes({
      quotationMarks: !quotationMarks
    })
  }))), createElement(BaseControl, null, createElement(ControlWrapper, {
    orientation: "vertical"
  }, createElement(ControlLabel, {
    label: "Button text"
  }), createElement(TextControl, {
    placeholder: "Button Text",
    value: buttonText,
    onChange: val => setAttributes({
      buttonText: val
    })
  }))));
};
export default statementControls;
