// @quillforms/admin-components/build/block-types-list-dropdown/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _components = require("@wordpress/components");
var _emotion = require("emotion");
var _blockTypesList = _interopRequireDefault(require("../block-types-list"));
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/**
 * WordPress Dependencies
 */

/**
 * External Dependencies
 */

/**
 * Internal Dependencies
 */

const BlockTypesListDropdown = ({
  destinationIndex,
  color = 'primary',
  parent
}) => {
  return (0, _react.createElement)(_components.Dropdown, {
    position: "bottom right",
    className: (0, _emotion.css)`
				.components-popover__content {
					width: 250px;
					max-height: 350px !important;
					overflow-y: hidden !important;

					& > div {
						padding: 0 !important;
					}
				}
			`,
    renderToggle: ({
      isOpen,
      onToggle
    }) => (0, _react.createElement)("div", {
      onClick: onToggle,
      "aria-expanded": isOpen,
      className: (0, _emotion.css)`
						margin: 0 10px;
						background: ${color === 'primary' ? `#3a3a3a` : '#fff'};
						cursor: pointer;
						width: 25px;
						height: 25px;
						display: flex;
						align-items: center;
						justify-content: center;
						border-radius: 50%;
						span {
							color: ${color === 'primary' ? `#fff` : `#333`};
							font-size: inherit;
							width: auto;
							height: auto;
						}
					`
    }, (0, _react.createElement)(_components.Icon, {
      icon: "plus"
    })),
    renderContent: () => (0, _react.createElement)(_react.Fragment, null, (0, _react.createElement)("p", {
      className: (0, _emotion.css)`
							font-weight: bold;
							padding: 0 15px;
							user-select: none;
						`
    }, "Choose a question type"), (0, _react.createElement)(_blockTypesList.default, {
      parent: parent,
      destinationIndex: destinationIndex
    }))
  });
};
var _default = exports.default = BlockTypesListDropdown;
