// @quillforms/renderer-core/build/components/field-render/context.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FieldRenderContextProvider = void 0;
exports.__experimentalUseFieldRenderContext = __experimentalUseFieldRenderContext;
var _react = require("react");
var _element = require("@wordpress/element");
var _lodash = require("lodash");
/**
 * QuillForms dependencies
 */

/**
 * WordPress Dependencies
 */

/**
 * External Dependencies
 */

const Context = (0, _element.createContext)({
  id: undefined,
  blockName: undefined,
  attributes: undefined,
  isActive: false,
  shouldBeRendered: false,
  isSubmitBtnVisible: false,
  isErrMsgVisible: false,
  showErrMsg: _lodash.noop,
  showNextBtn: _lodash.noop,
  next: _lodash.noop,
  isLastField: false
});
const {
  Provider
} = Context;
const FieldRenderContextProvider = ({
  value,
  children
}) => {
  return (0, _react.createElement)(Provider, {
    value: (0, _element.useMemo)(() => value, Object.values(value))
  }, children);
};

/**
 * A hook that returns the block render context.
 *
 * @return {Object} Block render context
 */
exports.FieldRenderContextProvider = FieldRenderContextProvider;
function __experimentalUseFieldRenderContext() {
  return (0, _element.useContext)(Context);
}
