// @quillforms/blocklib-short-text-block/build-module/admin/controls.js
import { createElement } from "react";
/**
 * QuillForms Dependencies
 */
import { BaseControl, ControlWrapper, ControlLabel, ToggleControl, TextControl } from '@quillforms/admin-components';

/**
 * WordPress Dependencies
 */
import { Fragment } from 'react';
const shortTextControl = ({
  attributes,
  setAttributes
}) => {
  const {
    setMaxCharacters,
    maxCharacters,
    minCharacters
  } = attributes;
  return createElement(Fragment, null, createElement(BaseControl, null, createElement(ControlWrapper, {
    orientation: "horizontal"
  }, createElement(ControlLabel, {
    label: "Min Characters",
    isNew: true
  }), createElement(ToggleControl, {
    checked: minCharacters !== false,
    onChange: () => {
      setAttributes({
        minCharacters: minCharacters !== false ? false : 0
      });
    }
  })), minCharacters !== false && createElement(TextControl, {
    type: "number",
    value: minCharacters,
    onChange: val => {
      setAttributes({
        minCharacters: val
      });
    }
  })), createElement(BaseControl, null, createElement(ControlWrapper, {
    orientation: "horizontal"
  }, createElement(ControlLabel, {
    label: "Max Characters"
  }), createElement(ToggleControl, {
    checked: setMaxCharacters,
    onChange: () => {
      setAttributes({
        setMaxCharacters: !setMaxCharacters
      });
    }
  })), setMaxCharacters && createElement(TextControl, {
    type: "number",
    value: maxCharacters,
    onChange: val => setAttributes({
      maxCharacters: val
    })
  })));
};
export default shortTextControl;
