// @quillforms/admin-components/build/logic-conditions/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _data = require("@wordpress/data");
var _components = require("@wordpress/components");
var _icons = require("@wordpress/icons");
var _lodash = require("lodash");
var _comboboxControl = _interopRequireDefault(require("../combobox-control"));
var _operatorSelector = _interopRequireDefault(require("./operator-selector"));
var _textControl = _interopRequireDefault(require("../text-control"));
var _button = _interopRequireDefault(require("../button"));
var _advancedDropdown = _interopRequireDefault(require("./advanced-dropdown"));
/**
 * QuillForms Dependencies
 */

/**
 * WordPress Dependencies
 */

/**
 * External Dependencies
 */

/**
 * Internal Dependencies
 */

const LogicConditions = ({
  value,
  onChange,
  combobox
}) => {
  const {
    formBlocks,
    blockTypes
  } = (0, _data.useSelect)(select => {
    return {
      // @ts-ignore
      formBlocks: select('quillForms/block-editor').getBlocks(),
      blockTypes: select('quillForms/blocks').getBlockTypes()
    };
  });
  const blocks = [];
  if ((0, _lodash.size)(formBlocks) > 0) {
    (0, _lodash.forEach)(formBlocks, block => {
      if (blockTypes[block.name]?.supports?.editable) {
        blocks.push({
          ...block
        });
      }
      if (blockTypes[block.name]?.supports?.innerBlocks) {
        if ((0, _lodash.size)(block?.innerBlocks) > 0) {
          (0, _lodash.forEach)(block.innerBlocks, childBlock => {
            blocks.push({
              ...childBlock
            });
          });
        }
      }
    });
  }
  // update value on mount if input isn't array or is empty array.
  (0, _react.useEffect)(() => {
    if (!Array.isArray(value) || value.length === 0) {
      onChange([[{
        vars: [{}, {}]
      }]]);
    }
  }, []);
  const customize = combobox?.customize?.override ? combobox.customize.handler : customizable => {
    let {
      sections,
      options
    } = customizable;
    sections = sections.filter(section => ['fields', 'variables', 'hidden_fields'].includes(section.key));
    options = options.filter(option => {
      if (option.type === 'field') {
        var _option$other$name;
        const blockType = blockTypes[(_option$other$name = option.other?.name) !== null && _option$other$name !== void 0 ? _option$other$name : ''];
        return blockType?.supports?.logic && blockType?.supports?.logicConditions;
      } else if (['variable', 'hidden_field'].includes(option.type)) {
        return true;
      }
      return false;
    });
    if (combobox?.customize) {
      const customized = combobox.customize.handler({
        sections,
        options
      });
      sections = customized.sections;
      options = customized.options;
    }
    return {
      sections,
      options
    };
  };
  if (!Array.isArray(value)) {
    return null;
  }
  return (0, _react.createElement)("div", {
    className: "logic-conditions"
  }, value.map((group, gi) => (0, _react.createElement)("div", {
    key: gi
  }, (0, _react.createElement)("div", {
    className: "logic-conditions__group"
  }, value.length > 1 && (0, _react.createElement)("div", {
    className: "group-remove"
  }, (0, _react.createElement)(_components.Icon, {
    icon: _icons.closeSmall,
    onClick: () => {
      const $value = [...value];
      $value.splice(gi, 1);
      onChange($value);
    }
  })), group.map((condition, ci) => {
    var _combobox$excerptLeng;
    const var0Type = condition.vars[0].type;
    const var0Value = condition.vars[0].value;
    const op = condition.op;
    const var1Value = condition.vars[1].value;

    // some properties for field type.
    let block;
    let blockType;
    if (var0Type === 'field') {
      block = blocks.find($block => $block.id === condition.vars[0].value);
      blockType = blockTypes[block?.name];
      if (!blockType) {
        return null;
      }
    }

    // possible operators list.
    let operators = null;
    if (var0Type === 'field') {
      if (blockType.logicalOperators) {
        operators = blockType.logicalOperators;
      }
    } else if (var0Type === 'variable') {
      operators = ['is', 'is_not', 'lower_than', 'greater_than'];
    }
    if (!operators) {
      operators = ['is', 'is_not', 'starts_with', 'ends_with', 'contains', 'not_contains'];
    }
    let row1 = null;
    let row2 = null;

    // row1 render.
    row1 = (0, _react.createElement)("div", {
      className: "condition-row-1"
    }, (0, _react.createElement)("div", {
      className: "condition-var0"
    }, (0, _react.createElement)(_comboboxControl.default, {
      excerptLength: (_combobox$excerptLeng = combobox?.excerptLength) !== null && _combobox$excerptLeng !== void 0 ? _combobox$excerptLeng : 30,
      value: condition.vars[0],
      onChange: var0 => {
        const $value = [...value];
        $value[gi][ci] = {
          vars: [var0, {
            type: undefined
          }]
        };
        onChange($value);
      },
      isToggleEnabled: false,
      customize: customize,
      hideChooseOption: true,
      selectFirstOption: true
    })), (0, _react.createElement)("div", {
      className: "condition-op"
    }, var0Value && (0, _react.createElement)(_operatorSelector.default, {
      operators: operators,
      value: op !== null && op !== void 0 ? op : null,
      onChange: op => {
        const $value = [...value];
        $value[gi][ci].op = op;
        onChange($value);
      }
    })));

    // row2 render.
    if (var0Value) {
      row2 = (0, _react.createElement)("div", {
        className: "condition-row-2"
      }, var0Type === 'field' && blockType.logicControl ? (0, _react.createElement)(blockType.logicControl, {
        attributes: block.attributes,
        value: var1Value,
        setValue: var1 => {
          const $value = [...value];
          $value[gi][ci].vars[1].value = var1;
          onChange($value);
        },
        removeCondition: () => {
          const $value = [...value];
          $value[gi].splice(ci, 1);
          onChange($value);
        }
      }) : var0Type === 'variable' ?
      // <TextControl
      // 	type="number"
      // 	value={ var1Value ?? '' }
      // 	onChange={ ( var1 ) => {
      // 		const $value = [ ...value ];
      // 		$value[ gi ][
      // 			ci
      // 		].vars[ 1 ].value = var1;
      // 		onChange( $value );
      // 	} }
      // />
      (0, _react.createElement)(_advancedDropdown.default, {
        var0: condition.vars[0],
        var1: condition.vars[1],
        onChange: var1 => {
          const $value = [...value];
          $value[gi][ci].vars[1].value = var1.value;
          $value[gi][ci].vars[1].type = var1.type;
          onChange($value);
        }
      }) : (0, _react.createElement)(_textControl.default, {
        value: var1Value !== null && var1Value !== void 0 ? var1Value : '',
        onChange: var1 => {
          const $value = [...value];
          $value[gi][ci].vars[1].value = var1;
          onChange($value);
        }
      }));
    }
    return (0, _react.createElement)("div", {
      key: ci
    }, (0, _react.createElement)("div", {
      className: "logic-conditions__condition"
    }, (0, _react.createElement)("div", {
      className: "condition-rows"
    }, row1, row2), group.length > 1 && (0, _react.createElement)("div", null, (0, _react.createElement)("div", {
      className: "condition-remove"
    }, (0, _react.createElement)(_components.Icon, {
      icon: _icons.closeSmall,
      onClick: () => {
        const $value = [...value];
        $value[gi].splice(ci, 1);
        onChange($value);
      }
    })))), group.length - 1 !== ci ? (0, _react.createElement)("div", null, "and") : (0, _react.createElement)("div", {
      className: "logic-conditions__add-condition"
    }, (0, _react.createElement)(_button.default, {
      isPrimary: true,
      onClick: () => {
        const $value = [...value];
        $value[gi].push({
          vars: [{}, {}]
        });
        onChange($value);
      }
    }, "and")));
  })), value.length - 1 !== gi ? (0, _react.createElement)("div", {
    className: "logic-conditions__or"
  }, "- OR -") : (0, _react.createElement)("div", {
    className: "logic-conditions__add-group"
  }, (0, _react.createElement)(_button.default, {
    isPrimary: true,
    onClick: () => {
      const $value = [...value];
      $value.push([{
        vars: [{}, {}]
      }]);
      onChange($value);
    }
  }, "OR")))));
};
var _default = exports.default = LogicConditions;
