// @quillforms/blocklib-slider-block/build/renderer/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.rendererSettings = void 0;
var _blocks = require("@quillforms/blocks");
var _display = _interopRequireDefault(require("./display"));
/**
 * QuillForms Dependencies
 */
/**
 * Internal Dependencies
 */
const {
  name: name
} = {
  name: "slider",
  supports: {
    editable: true,
    required: true,
    attachment: true,
    description: true,
    logic: true,
    defaultValue: true,
    theme: true,
    numeric: true,
    payments: true
  },
  attributes: {
    min: {
      type: "number",
      "default": 0
    },
    max: {
      type: "number",
      "default": 100
    },
    step: {
      type: "number",
      "default": 10
    },
    prefix: {
      type: "string",
      "default": ""
    },
    suffix: {
      type: "string",
      "default": ""
    },
    marks: {
      type: "string",
      "default": "no"
    },
    customMarks: {
      type: "array",
      "default": [{
        label: "",
        value: ""
      }]
    }
  },
  logicalOperators: ["is", "is_not", "greater_than", "lower_than"]
};
const rendererSettings = exports.rendererSettings = {
  display: _display.default
};
(0, _blocks.setBlockRendererSettings)(name, rendererSettings);
