// @quillforms/renderer-core/build/components/payment-modal/info/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _utils = require("@quillforms/utils");
var _useGeneralTheme = _interopRequireDefault(require("../../../hooks/use-general-theme"));
var _config = _interopRequireDefault(require("@quillforms/config"));
var _i18n = require("@wordpress/i18n");
var _data = require("@wordpress/data");
var _element = require("@wordpress/element");
var _emotion = require("emotion");
var _classnames = _interopRequireDefault(require("classnames"));
var _tinycolor = _interopRequireDefault(require("tinycolor2"));
var _reactLoaderSpinner = require("react-loader-spinner");
/**
 * QuillForms Dependencies
 */

/**
 * WordPress Dependencies
 */

/**
 * External Dependencies
 */

const Info = ({
  data
}) => {
  var _data$payments$labels, _data$payments$labels2;
  const generalTheme = (0, _useGeneralTheme.default)();
  const questionsColor = (0, _tinycolor.default)(generalTheme.questionsColor);
  const cs = data.payments.currency.symbol;
  const csp = data.payments.currency.symbol_pos;
  const total = data.payments?.discount_details ? data.payments.discount_details.amount : data.payments.products.total;
  const {
    setPaymentData
  } = (0, _data.useDispatch)('quillForms/renderer-core');
  const [isDeleting, setIsDeleting] = (0, _element.useState)(false);
  const deleteCoupon = async () => {
    setIsDeleting(true);
    try {
      const {
        submission_id,
        hashed_id
      } = data;
      let response = await fetch(_config.default.getAdminUrl() + 'admin-ajax.php', {
        method: 'POST',
        body: new URLSearchParams({
          action: 'quillforms_delete_coupon',
          submissionId: submission_id,
          hashedId: hashed_id,
          coupon: data.payments.discount_details.coupon
        })
      });
      let result = await response.json();
      if (result.success) {
        // Update new products
        const UpdatedData = {
          ...data
        };
        delete UpdatedData.payments.discount_details;
        setPaymentData(UpdatedData);
      }
    } catch (e) {
      console.log('deleteCoupon: error throwed', e);
      return {
        success: false,
        message: e instanceof Error && e.message ? e.message : 'Unexpected error'
      };
    }
    setIsDeleting(false);
  };
  return (0, _react.createElement)("div", {
    className: "renderer-core-payment-modal-info"
  }, (0, _react.createElement)("div", {
    className: (0, _classnames.default)('renderer-core-payment-modal-info__heading', (0, _emotion.css)`
						font-size: 22px;
						text-transform: uppercase;
						background: ${generalTheme.buttonsBgColor};
						padding: 15px;
						color: ${generalTheme.buttonsFontColor};
						border-top-right-radius: 8px;
						border-top-left-radius: 8px;
					`)
  }, (_data$payments$labels = data.payments.labels?.order_details_heading) !== null && _data$payments$labels !== void 0 ? _data$payments$labels : 'Your Order'), data.payments.products.items.map((item, index) => {
    if (item.price) {
      return (0, _react.createElement)("div", {
        key: index,
        className: (0, _classnames.default)('renderer-core-payment-order__item-row', (0, _emotion.css)`
									color: ${generalTheme.questionsColor};
									border-bottom: 1px solid;
									border-color: ${questionsColor.setAlpha(0.3).toString()};
								`)
      }, (0, _react.createElement)("div", {
        className: "renderer-core-payment-order__item-name"
      }, item.name), (0, _react.createElement)("div", {
        className: "renderer-core-payment-order__item-price"
      }, (0, _utils.formatMoney)(item.price, cs, csp), ' '));
    } else {
      return (0, _react.createElement)(_react.Fragment, null);
    }
  }), data.payments?.discount_details && (0, _react.createElement)("div", {
    className: (0, _classnames.default)('renderer-core-payment-modal-info_discount', (0, _emotion.css)`
							color: ${generalTheme.questionsColor};
							border-bottom: 1px solid;
							border-color: ${questionsColor.setAlpha(0.3).toString()};
							font-weight: bold;
						`)
  }, (0, _react.createElement)("div", null, (0, _i18n.__)('Discount', 'quillforms'), " "), (0, _react.createElement)("div", {
    className: (0, _classnames.default)('renderer-core-payment-modal-info_discount__details', (0, _emotion.css)`
								display: inline-flex;
								align-items: center;
							`)
  }, (0, _react.createElement)("span", {
    className: "renderer-core-payment-modal-info_discount__code"
  }, data.payments.discount_details.code), (0, _react.createElement)("span", {
    style: {
      margin: '0 5px'
    }
  }, "-"), (0, _utils.formatMoney)(data.payments.discount_details.discount, cs, csp), data.payments.discount_details.type === 'percent' && (0, _react.createElement)("span", {
    className: "renderer-core-payment-modal-info_discount__type",
    style: {
      marginLeft: '5px'
    }
  }, `(${data.payments.discount_details.discount_amount}%)`), (0, _react.createElement)("div", {
    className: (0, _classnames.default)('renderer-core-payment-modal-info_delete-discount', (0, _emotion.css)`
									display: inline-block;
									margin-left: 10px;
									color: #a94442;
									&:hover {
										cursor: pointer;
									}
								`),
    onClick: deleteCoupon
  }, isDeleting ? (0, _react.createElement)(_reactLoaderSpinner.TailSpin, {
    color: '#a94442',
    height: 20,
    width: 20
  }) : (0, _react.createElement)("svg", {
    fill: "currentColor",
    height: "20",
    width: "20",
    viewBox: "0 0 1536 1792"
  }, (0, _react.createElement)("path", {
    d: "M1149 1122q0-26-19-45l-181-181 181-181q19-19 19-45 0-27-19-46l-90-90q-19-19-46-19-26 0-45 19l-181 181-181-181q-19-19-45-19-27 0-46 19l-90 90q-19 19-19 46 0 26 19 45l181 181-181 181q-19 19-19 45 0 27 19 46l90 90q19 19 46 19 26 0 45-19l181-181 181 181q19 19 45 19 27 0 46-19l90-90q19-19 19-46zM1536 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
  }))))), (0, _react.createElement)("div", {
    className: (0, _classnames.default)('renderer-core-payment-modal-info_footer', (0, _emotion.css)`
						color: ${generalTheme.questionsColor};
						font-weight: bold;
					`)
  }, (0, _react.createElement)("div", null, (_data$payments$labels2 = data.payments.labels?.order_total) !== null && _data$payments$labels2 !== void 0 ? _data$payments$labels2 : 'Total', " "), (0, _react.createElement)("div", null, (0, _utils.formatMoney)(total, cs, csp), ' ', data.payments.recurring && ` /${data.payments.recurring.interval_count} ${data.payments.recurring.interval_unit}/s`)));
};
var _default = exports.default = Info;
