// @quillforms/blocklib-multiple-choice-block/build-module/admin/controls.js
import { createElement } from "react";
/**
 * QuillForms Dependencies
 */
import { BaseControl, ControlWrapper, ControlLabel, ToggleControl, ChoicesBulkBtn, ChoicesInserter, TextControl } from '@quillforms/admin-components';

/**
 * WordPress Dependencies
 */
import { Fragment } from 'react';
const multipleChoiceControls = props => {
  const {
    attributes: {
      multiple,
      verticalAlign,
      choices,
      min,
      max
    },
    setAttributes
  } = props;
  return createElement(Fragment, null, createElement(BaseControl, null, createElement(ControlWrapper, {
    orientation: "horizontal"
  }, createElement(ControlLabel, {
    label: "Multiple"
  }), createElement(ToggleControl, {
    checked: multiple,
    onChange: () => setAttributes({
      multiple: !multiple
    })
  }))), multiple && createElement(Fragment, null, createElement(BaseControl, null, createElement(ControlWrapper, {
    orientation: "horizontal"
  }, createElement(ControlLabel, {
    label: "Minimum limit for choices",
    isNew: true
  }), createElement(ToggleControl, {
    checked: min,
    onChange: () => {
      setAttributes({
        min: min === false ? 1 : false
      });
    }
  })), min !== false && createElement(TextControl, {
    type: "number",
    value: min,
    onChange: val => {
      setAttributes({
        min: val
      });
    }
  })), createElement(BaseControl, null, createElement(ControlWrapper, {
    orientation: "horizontal"
  }, createElement(ControlLabel, {
    label: "Maximum limit for choices",
    isNew: true
  }), createElement(ToggleControl, {
    checked: max,
    onChange: () => {
      setAttributes({
        max: max === false ? 5 : false
      });
    }
  })), max !== false && createElement(TextControl, {
    type: "number",
    value: max,
    onChange: val => {
      setAttributes({
        max: val
      });
    }
  }))), createElement(BaseControl, null, createElement(ControlWrapper, {
    orientation: "horizontal"
  }, createElement(ControlLabel, {
    label: "Vertical Align"
  }), createElement(ToggleControl, {
    checked: verticalAlign,
    onChange: () => setAttributes({
      verticalAlign: !verticalAlign
    })
  }))), createElement(BaseControl, null, createElement(ControlWrapper, {
    orientation: "horizontal"
  }, createElement(ControlLabel, {
    label: "Choices"
  }), createElement(ChoicesBulkBtn, {
    choices: choices,
    setChoices: val => {
      setAttributes({
        choices: val
      });
    }
  })), createElement(ControlWrapper, {
    orientation: "vertical"
  }, createElement(ChoicesInserter, {
    choices: choices,
    setChoices: val => {
      setAttributes({
        choices: val
      });
    }
  }))));
};
export default multipleChoiceControls;
