// @quillforms/blocks/build/store/actions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.setBlockRendererSettings = exports.setBlockAdminSettings = exports.addBlockTypes = void 0;
var _constants = require("./constants");
var _lodash = require("lodash");
/**
 * Returns an action object used in registering block renderer settings.
 *
 * @param  blockRendererSettings Block renderer settings which define block renderer behavior.
 * @param  name                  The block name.
 *
 * @return Action object.
 */
const setBlockRendererSettings = (blockRendererSettings, name) => {
  return {
    type: _constants.SET_BLOCK_RENDERER_SETTINGS,
    settings: blockRendererSettings,
    name
  };
};

/**
 * Returns an action object used in registering block editor configuration.
 *
 * @param  blockAdminSettings Block admin settings which defines block behavior at admin.
 * @param  name               The block name.
 *
 * @return Action object.
 */
exports.setBlockRendererSettings = setBlockRendererSettings;
const setBlockAdminSettings = (blockAdminSettings, name) => {
  return {
    type: _constants.SET_BLOCK_ADMIN_SETTINGS,
    settings: blockAdminSettings,
    name
  };
};

/**
 * Returns an action object used in signalling that block types have been added.
 *
 * @param  blockTypes Block types received.
 *
 * @return Action object.
 */
exports.setBlockAdminSettings = setBlockAdminSettings;
const addBlockTypes = blockTypes => {
  return {
    type: _constants.ADD_BLOCK_TYPES,
    blockTypes: (0, _lodash.castArray)(blockTypes)
  };
};
exports.addBlockTypes = addBlockTypes;
