// @quillforms/blocks/build/store/reducer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _lodash = require("lodash");
var _constants = require("./constants");
/**
 * External dependencies
 */

/**
 * Internal Dependencies
 */

const initialState = {
  unknown: {
    title: 'Unknown',
    supports: {
      editable: true,
      description: true,
      attachment: true,
      required: false,
      logic: true,
      placeholder: false
    },
    logicalOperators: ['is', 'is_not', 'starts_with', 'greater_than', 'lower_than', 'ends_with', 'contains', 'not_contains']
  }
};

/**
 * Reducer returning an array of registered blocks.
 *
 * @param {Object} state  Current state.
 * @param {Object} action Dispatched action.
 *
 * @return {Object} Updated state.
 */
const BlocksReducer = (state = initialState, action) => {
  switch (action.type) {
    case _constants.SET_BLOCK_ADMIN_SETTINGS:
    case _constants.SET_BLOCK_RENDERER_SETTINGS:
      const {
        name
      } = action;
      if (!state[name]) {
        return state;
      }
      return {
        ...state,
        [name]: {
          ...state[name],
          ...action.settings
        }
      };
    case _constants.ADD_BLOCK_TYPES:
      {
        return {
          ...state,
          ...(0, _lodash.keyBy)(action.blockTypes, 'name')
        };
      }
  }
  return state;
};
var _default = exports.default = BlocksReducer;
