// @quillforms/admin-components/build/choices-inserter/choice-wrapper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _draggable = _interopRequireDefault(require("../draggable"));
var _choiceRow = _interopRequireDefault(require("./choice-row"));
/**
 * QuillForms Dependencies
 */

const ChoiceWrapper = ({
  index,
  style,
  data
}) => {
  const item = data[index];

  // We are rendering an extra item for the placeholder
  if (item === undefined) {
    return null;
  }
  return (0, _react.createElement)(_draggable.default, {
    key: item.value,
    draggableId: item.value,
    index: index
  }, (provided, snapshot) => (0, _react.createElement)("div", {
    ...provided.draggableProps,
    ref: provided.innerRef,
    "date-isDragging": snapshot.isDragging,
    style: {
      margin: 0,
      ...provided.draggableProps.style,
      ...style
    }
  }, (0, _react.createElement)(_choiceRow.default, {
    choices: data,
    index: index,
    provided: provided
  })));
};
var _default = exports.default = ChoiceWrapper;
