// @quillforms/renderer-core/build/components/form-render/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _blocks = require("@quillforms/blocks");
var _utils = require("@quillforms/utils");
var _element = require("@wordpress/element");
var _hooks = require("@wordpress/hooks");
var _formContext = require("../form-context");
var _formWrapper = _interopRequireDefault(require("../form-wrapper"));
var _lodash = require("lodash");
/**
 * QuillForms Dependencies
 */

/**
 * WordPress Dependencies
 */

/**
 * Internal Dependencies
 */

/**
 * External Dependencies
 */

const Form = ({
  formObj,
  formId,
  onSubmit,
  applyLogic = false,
  beforeGoingNext,
  isPreview = false,
  customFonts
}) => {
  const [deviceWidth, setDeviceWidth] = (0, _element.useState)('');
  (0, _element.useEffect)(() => {
    function handleResize() {
      if (window.innerWidth < 600) {
        setDeviceWidth('mobile');
      } else {
        setDeviceWidth('desktop');
      }
    }
    window.addEventListener('resize', handleResize);
    handleResize();
    return () => window.removeEventListener('resize', handleResize);
  }, []);
  const formatFormObj = () => {
    // If not in preview mode, sanitize blocks.
    // In preview mode, sanitizing is already done in block editor resolvers.
    if (!isPreview) {
      formObj.blocks = (0, _blocks.sanitizeBlocks)(formObj.blocks);
    }
    formObj.messages = {
      ...(0, _utils.getDefaultMessages)(),
      ...formObj.messages
    };
    const ParsedUrlSearch = typeof window !== 'undefined' ? new URLSearchParams(window?.location?.search?.substring(1)) : '';
    if (!formObj.settings) formObj.settings = {};
    formObj.settings = {
      disableProgressBar: false,
      disableWheelSwiping: false,
      disableNavigationArrows: false,
      animationDirection: 'vertical',
      showQuestionsNumbers: true,
      showLettersOnAnswers: true,
      saveAnswersInBrowser: false,
      displayBranding: false,
      ...formObj.settings
    };
    formObj.correctIncorrectQuiz = {
      enabled: false,
      questions: {},
      showAnswersDuringQuiz: true,
      ...formObj.correctIncorrectQuiz
    };
    formObj.blocks = (0, _lodash.map)(formObj.blocks, $block => {
      if (($block.name === 'multiple-choice' || $block.name === 'dropdown' || $block.name === 'picture-choice') && $block?.attributes?.randomize && !isPreview) {
        var _$block$attributes$ch;
        return {
          ...$block,
          attributes: {
            ...$block.attributes,
            // @ts-expect-error
            choices: (0, _lodash.shuffle)((_$block$attributes$ch = $block?.attributes?.choices) !== null && _$block$attributes$ch !== void 0 ? _$block$attributes$ch : [])
          }
        };
      }
      return $block;
    });

    // 'quillforms-redirection' is deprecated and will be removed.
    if ((0, _lodash.size)(ParsedUrlSearch) > 0 &&
    // @ts-expect-error
    ParsedUrlSearch?.get('quillforms-shortcode')) {
      formObj.settings.disableWheelSwiping = true;
    }
    return formObj;
  };
  (0, _element.useEffect)(() => {
    if (!isPreview) {
      (0, _hooks.doAction)('QuillForms.RendererCore.Loaded');
    }
  }, []);
  return (0, _react.createElement)(_formContext.FormContextProvider, {
    value: {
      formObj: formatFormObj(),
      onSubmit,
      isPreview,
      formId,
      beforeGoingNext,
      deviceWidth,
      customFonts
    }
  }, (0, _react.createElement)(_formWrapper.default, {
    applyLogic: applyLogic
  }));
};
var _default = exports.default = Form;
