// @quillforms/admin-components/build-module/choices-bulk-btn/index.js
import { createElement } from "react";
/**
 * WordPress Dependencies
 */
import { useState } from 'react';

/**
 * External Dependencies
 */
import { css } from 'emotion';

/**
 * Internal Dependencies
 */
import Button from '../button';
import BulkChoicesModal from './modal';
import generateChoiceId from '../choices-inserter/generate-choice-id';
const ChoicesBulkBtn = ({
  choices,
  setChoices
}) => {
  const [modalOpen, setModalOpen] = useState(false);
  const insertBulkChoices = txt => {
    const $choices = [...choices];
    txt.trim().split('\n').forEach(item => {
      $choices.push({
        value: generateChoiceId(),
        label: item
      });
    });
    setChoices($choices);
  };
  return createElement("div", {
    className: "admin-components-choices-bulk-btn"
  }, createElement(Button, {
    isDefault: true,
    className: css`
					display: inline-block;
					background: #000;
					color: #fff;
					padding: 5px;
					margin-top: 12px;
				`,
    onClick: () => setModalOpen(true)
  }, "Bulk Answers"), modalOpen && createElement(BulkChoicesModal, {
    onInsert: txt => insertBulkChoices(txt),
    onCloseModal: () => setModalOpen(false)
  }));
};
export default ChoicesBulkBtn;
