// @quillforms/admin-components/build-module/choices-bulk-btn/modal.js
import { createElement, Fragment } from "react";
/**
 * WordPress Dependencies.
 */
import { Modal } from '@wordpress/components';
import { useRef, useState, useEffect } from 'react';
/**
 * External Dependencies
 */
import { css } from 'emotion';
import classnames from 'classnames';
import { Oval as Loader } from 'react-loader-spinner';

/**
 * Internal Dependencies
 */
import Button from '../button';
const ChoicesBulkModal = ({
  onCloseModal,
  onInsert
}) => {
  const [isInserting, setIsInserting] = useState(false);
  const [bulkChoicesTxt, setBulkChoicesTxt] = useState('');
  const ref = useRef(null);
  useEffect(() => {
    if (ref?.current) {
      ref.current.focus();
    }
  });
  return createElement(Modal, {
    className: classnames('admin-components-choices-bulk-modal', css`
					border: none !important;
					min-width: 350px !important;
					border-radius: 10px;
					z-index: 1111111;

					.components-modal__content {
						display: flex;
						flex-direction: column;
						justify-content: center;
						background: #eee;
						margin-top: 54px;
						min-height: 300px;
					}

					.components-modal__header {
						background: #313e57;
						.components-modal__header-heading {
							color: #fff;
						}
						.components-button.has-icon svg {
							fill: #fff;
						}
					}
				`),
    title: "Bulk Choices",
    onRequestClose: onCloseModal
  }, createElement("p", {
    className: css`
					marging-bottom: 10px;
				`
  }, createElement("strong", null, "Insert each answer choice in a separate line ")), createElement("textarea", {
    ref: ref,
    className: css`
					width: 100%;
					height: 170px;
					overflow-y: auto;
					resize: none;
				`,
    onChange: e => setBulkChoicesTxt(e.target.value),
    value: bulkChoicesTxt
  }), createElement("div", {
    className: css`
					display: flex;
					margin-top: 10px;
					justify-content: flex-end;
				`
  }, createElement(Button, {
    isDefault: true,
    isLarge: true,
    className: css`
						margin-right: 10px !important;
					`,
    onClick: onCloseModal
  }, "Cancel"), createElement(Button, {
    isLarge: true,
    className: css`
						width: 70px;
						display: flex;
						justify-content: center;
						align-items: center;
					`,
    onClick: () => {
      if (isInserting) return;
      setIsInserting(true);
      onInsert(bulkChoicesTxt);
      setTimeout(() => {
        setIsInserting(false);
        onCloseModal();
      }, 0);
    },
    isPrimary: true
  }, "Done", createElement(Fragment, null, isInserting && createElement("div", {
    className: css`
									display: flex;
									justify-content: center;
									align-items: center;
									margin: 0px 10px;
								`
  }, createElement(Loader, {
    color: "#00BFFF",
    height: 15,
    width: 15
  }))))));
};
export default ChoicesBulkModal;
