// @quillforms/admin-components/build/share-form-modal/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _data = require("@wordpress/data");
var _apiFetch = _interopRequireDefault(require("@wordpress/api-fetch"));
var _url = require("@wordpress/url");
var _components = require("@wordpress/components");
var _emotion = require("emotion");
var _classnames = _interopRequireDefault(require("classnames"));
var _linkPlaceholder = _interopRequireDefault(require("./link-placeholder"));
/**
 * WordPress Dependencies
 */

/**
 * External Dependencies
 */

/**
 * Internal Dependencies
 */

const ShareFormModal = ({
  formId,
  closeModal
}) => {
  const [isFetching, setIsFetching] = (0, _react.useState)(true);
  const [permalink, setPermalink] = (0, _react.useState)('');
  const {
    hiddenFields
  } = (0, _data.useSelect)(select => {
    var _select$getHiddenFiel;
    return {
      hiddenFields: (_select$getHiddenFiel = select('quillForms/hidden-fields-editor'
      //@ts-expect-error
      )?.getHiddenFields()) !== null && _select$getHiddenFiel !== void 0 ? _select$getHiddenFiel : []
    };
  });

  // Invalidate resolution for entity record on unmount
  (0, _react.useEffect)(() => {
    (0, _apiFetch.default)({
      path: (0, _url.addQueryArgs)(`/wp/v2/quill_forms/${formId}`, {
        _fields: ['link']
      })
    }).then(res => {
      setIsFetching(false);
      //@ts-expect-error
      setPermalink(res.link);
    });
  }, []);
  let link = permalink;
  if (hiddenFields.length > 0) {
    const query = {};
    const hash = {};
    for (const field of hiddenFields) {
      if (['utm_source', 'utm_medium', 'utm_campaign', 'utm_term', 'utm_content'].includes(field.name)) {
        query[field.name] = 'xxxxx';
      } else {
        hash[field.name] = 'xxxxx';
      }
    }
    if (Object.keys(query).length) {
      link += '?' + (0, _url.buildQueryString)(query);
    }
    if (Object.keys(hash).length) {
      link += '#' + (0, _url.buildQueryString)(hash);
    }
  }
  return (0, _react.createElement)(_components.Modal, {
    className: (0, _classnames.default)((0, _emotion.css)`
					border: none !important;
					min-width: 420px !important;
					max-width: 470px !important;
					border-radius: 10px;
					z-index: 1111111;
				`),
    title: "Share form",
    onRequestClose: closeModal
  }, (0, _react.createElement)("div", {
    className: (0, _classnames.default)('admin-components-share-form-modal__link', (0, _emotion.css)`
						display: flex;
						flex-direction: column;
						margin-top: 10px;
					`)
  }, (0, _react.createElement)("div", {
    className: (0, _emotion.css)`
						margin-bottom: 6px;
						font-weight: bold;
					`
  }, "Link"), isFetching ? (0, _react.createElement)(_linkPlaceholder.default, null) : (0, _react.createElement)("div", {
    className: (0, _emotion.css)`
							border-radius: 5px;
							padding: 8px;
							background: #eeeeee;
						`
  }, (0, _react.createElement)("a", {
    href: link,
    target: "_blank",
    rel: "noreferrer"
  }, link))), (0, _react.createElement)("div", {
    className: (0, _classnames.default)('admin-components-share-form-modal__shortcode', (0, _emotion.css)`
						display: flex;
						flex-direction: column;
						margin-top: 10px;
					`)
  }, (0, _react.createElement)("div", {
    className: (0, _emotion.css)`
						margin-bottom: 6px;
						font-weight: bold;
					`
  }, "Shortcode"), isFetching ? (0, _react.createElement)(_linkPlaceholder.default, null) : (0, _react.createElement)("div", {
    className: (0, _emotion.css)`
							border-radius: 5px;
							padding: 8px;
							background: #eeeeee;
						`
  }, `[quillforms id="${formId}" width="100%"]`)), (0, _react.createElement)("div", {
    className: (0, _classnames.default)('admin-components-share-form-modal__embed', (0, _emotion.css)`
						display: flex;
						flex-direction: column;
						margin-top: 10px;
					`)
  }, (0, _react.createElement)("div", {
    className: (0, _emotion.css)`
						margin-bottom: 6px;
						font-weight: bold;
					`
  }, "Embed"), isFetching ? (0, _react.createElement)(_linkPlaceholder.default, null) : (0, _react.createElement)("div", {
    className: (0, _emotion.css)`
							border-radius: 5px;
							padding: 8px;
							background: #eeeeee;
						`
  }, `<iframe src="${link}" width="100%" height="600" style="border:0;"></iframe>`)));
};
var _default = exports.default = ShareFormModal;
