// @quillforms/renderer-core/build/components/payment-modal/methods/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _paymentGateways = require("@quillforms/payment-gateways");
var _button = _interopRequireDefault(require("../../button"));
var _config = _interopRequireDefault(require("@quillforms/config"));
var _radioControl = _interopRequireDefault(require("./radio-control"));
var _element = require("@wordpress/element");
var _data = require("@wordpress/data");
var _i18n = require("@wordpress/i18n");
var _useGeneralTheme = _interopRequireDefault(require("../../../hooks/use-general-theme"));
var _emotion = require("emotion");
var _lodash = require("lodash");
var _classnames = _interopRequireDefault(require("classnames"));
var _reactLoaderSpinner = require("react-loader-spinner");
/**
 * QuillForms Dependencies
 */

/**
 * WordPress Dependencies
 */

/**
 * External Dependencies
 */

const Methods = ({
  data
}) => {
  var _urlParams$get, _data$payments$labels, _data$payments$labels2;
  const {
    setPaymentData,
    completeForm
  } = (0, _data.useDispatch)('quillForms/renderer-core');
  const generalTheme = (0, _useGeneralTheme.default)();
  const gateways = (0, _paymentGateways.getPaymentGatewayModules)();
  const methodsKeys = Object.keys(data.payments.methods);
  const options = methodsKeys.map(key => {
    const [gateway, method] = key.split(':');
    return {
      label: gateways[gateway].methods[method].customer.label.text,
      value: key
    };
  });
  const urlParams = new URLSearchParams(window.location.search);
  let defaultMethod = methodsKeys[0];
  if (urlParams.get('step') === 'payment' && methodsKeys.includes((_urlParams$get = urlParams.get('method')) !== null && _urlParams$get !== void 0 ? _urlParams$get : '')) {
    var _urlParams$get2;
    defaultMethod = (_urlParams$get2 = urlParams.get('method')) !== null && _urlParams$get2 !== void 0 ? _urlParams$get2 : '';
  }
  const [selected, setSelected] = (0, _element.useState)(defaultMethod);
  const [gateway, method] = selected.split(':');
  const CustomerRender = gateways[gateway].methods[method].customer.render;
  const discountDetails = data.payments?.discount_details;
  const discountAmount = discountDetails?.amount;
  const [isPaying, setIsPaying] = (0, _element.useState)(false);
  const completeFullDiscountedOrders = async () => {
    if (isPaying) return;
    setIsPaying(true);
    try {
      const {
        submission_id,
        hashed_id
      } = data;
      let response = await fetch(_config.default.getAdminUrl() + 'admin-ajax.php', {
        method: 'POST',
        body: new URLSearchParams({
          action: 'quillforms_complete_full_discounted_orders',
          submissionId: submission_id,
          hashedId: hashed_id
        })
      });
      let result = await response.json();
      if (result.success) {
        completeForm();
      } else {
        throw new Error(result.message);
      }
    } catch (e) {
      console.log('completePendingSubmission: error throwed', e);
      return {
        success: false,
        message: e instanceof Error && e.message ? e.message : 'Unexpected error'
      };
    }
    setIsPaying(false);
  };
  return (0, _react.createElement)("div", {
    className: "renderer-core-payment-modal-methods"
  }, (0, _react.createElement)(_react.Fragment, null, (0, _lodash.size)(methodsKeys) > 1 && (0, _react.createElement)(_react.Fragment, null, (0, _react.createElement)("p", {
    className: (0, _emotion.css)`
								color: ${generalTheme.questionsColor};
								font-size: 20px;
							`
  }, (_data$payments$labels = data.payments.labels?.select_payment_method) !== null && _data$payments$labels !== void 0 ? _data$payments$labels : 'Select a payment method'), (0, _react.createElement)("div", {
    className: "renderer-components-radio-control__options-group"
  }, (0, _react.createElement)(_radioControl.default, {
    id: "payment-methods",
    selected: selected,
    options: options,
    onChange: setSelected
  })))), discountAmount !== 0 && (0, _react.createElement)(CustomerRender, {
    slug: selected,
    data: data,
    onComplete: () => {
      completeForm();
      setPaymentData(null);
    }
  }), discountAmount === 0 && (0, _react.createElement)("div", null, (0, _react.createElement)(_button.default, {
    className: (0, _classnames.default)({
      loading: isPaying
    }, (0, _emotion.css)`
								&.loading .renderer-core-arrow-icon {
									display: none;
								}
							`, 'payment-button'),
    onClick: () => {
      completeFullDiscountedOrders();
    }
  }, (0, _react.createElement)("span", {
    id: "button-text"
  }, isPaying ? (0, _react.createElement)(_reactLoaderSpinner.TailSpin, {
    color: generalTheme.buttonsFontColor,
    height: 50,
    width: 50
  }) : (0, _react.createElement)(_react.Fragment, null, (_data$payments$labels2 = data?.payments?.labels?.pay) !== null && _data$payments$labels2 !== void 0 ? _data$payments$labels2 : (0, _i18n.__)('Pay now', 'quillforms'))))));
};
var _default = exports.default = Methods;
