// @quillforms/admin-components/build-module/rich-text/components/link/index.js
import { createElement } from "react";
/**
 * External Dependencies
 */
import { Transforms, Editor, Range } from 'slate';
/**
 * Inernal Dependencies
 */
import Button from '../button';
import LinkIcon from './link-icon';
const Link = ({
  editor
}) => {
  const insertLink = url => {
    if (editor.selection) {
      wrapLink(url);
    }
  };
  const isLinkActive = () => {
    const [link] = Editor.nodes(editor, {
      match: n => n.type === 'link'
    });
    return !!link;
  };
  const unwrapLink = () => {
    Transforms.unwrapNodes(editor, {
      match: n => n.type === 'link'
    });
  };
  const wrapLink = url => {
    if (isLinkActive()) {
      unwrapLink();
    }
    const {
      selection
    } = editor;
    const isCollapsed = selection && Range.isCollapsed(selection);
    const link = {
      type: 'link',
      url,
      children: isCollapsed ? [{
        text: url
      }] : []
    };
    if (isCollapsed) {
      Transforms.insertNodes(editor, link);
    } else {
      Transforms.wrapNodes(editor, link, {
        split: true
      });
      Transforms.collapse(editor, {
        edge: 'end'
      });
    }
  };
  return createElement(Button, {
    active: isLinkActive(),
    onMouseDown: event => {
      event.preventDefault();
      if (isLinkActive()) {
        unwrapLink();
        return;
      }
      // eslint-disable-next-line no-alert
      const url = window.prompt('Enter the URL of the link:');
      if (!url) return;
      insertLink(url);
    }
  }, createElement(LinkIcon, null));
};
export default Link;
