// @quillforms/renderer-core/build-module/components/payment-modal/coupon/index.js
import { createElement, Fragment } from "react";
/**
 * QuillForms Dependencies
 */
import useGeneralTheme from '../../../hooks/use-general-theme';
import configApi from '@quillforms/config';

/**
 * WordPress Dependencies
 */

import { useState } from '@wordpress/element';
import { useDispatch } from '@wordpress/data';

/**
 * External Dependencies
 */
import classnames from 'classnames';
import { TailSpin as Loader } from 'react-loader-spinner';
import { css } from 'emotion';
const Coupon = ({
  data
}) => {
  var _data$payments$labels, _data$payments$labels2, _data$payments$labels3;
  const generalTheme = useGeneralTheme();
  const [isPaying, setIsPaying] = useState(false);
  const [showCoupon, setShowCoupon] = useState(false);
  const [coupon, setCoupon] = useState('');
  const [error, setError] = useState('');
  const {
    setPaymentData
  } = useDispatch('quillForms/renderer-core');
  const applyCoupon = async () => {
    setError('');
    setIsPaying(true);
    try {
      const {
        submission_id,
        hashed_id
      } = data;
      let response = await fetch(configApi.getAdminUrl() + 'admin-ajax.php', {
        method: 'POST',
        body: new URLSearchParams({
          action: 'quillforms_apply_discount',
          submissionId: submission_id,
          hashedId: hashed_id,
          coupon
        })
      });
      let result = await response.json();
      if (result.success) {
        // Update new products
        const UpdatedData = {
          ...data
        };
        UpdatedData.payments.discount_details = result.data.details;
        UpdatedData.payments.discount_details['coupon'] = coupon;
        setPaymentData(UpdatedData);
      } else {
        setError(result.data);
      }
    } catch (e) {
      console.log('applyCoupon: error throwed', e);
      return {
        success: false,
        message: e instanceof Error && e.message ? e.message : 'Unexpected error'
      };
    }
    setIsPaying(false);
    setCoupon('');
  };
  return createElement("div", {
    className: "renderer-core-payment-coupon"
  }, createElement("div", {
    className: classnames("renderer-core-payment-coupon__label", css`
					color: ${generalTheme.questionsColor};
				`),
    onClick: () => setShowCoupon(true)
  }, (_data$payments$labels = data.payments.labels?.discountQuestion) !== null && _data$payments$labels !== void 0 ? _data$payments$labels : 'You have a coupon?'), showCoupon && createElement("div", {
    className: "render-core-payment-coupon-form"
  }, createElement("div", {
    className: "renderer-core-payment-coupon__input"
  }, createElement("input", {
    type: "text",
    placeholder: (_data$payments$labels2 = data.payments.labels?.discountPlaceholde) !== null && _data$payments$labels2 !== void 0 ? _data$payments$labels2 : 'Enter your discount code',
    style: {
      color: '#333',
      border: `1px solid rgba(0, 0, 0, 0.3)`,
      borderRadius: '8px',
      fontSize: '16px',
      padding: '10px 20px',
      lineHeight: '1',
      background: "#fff"
    },
    className: classnames(css`
									&:focus {
										outline: none;
										border: 1px solid rgba( 0, 0, 0, 0.3 );
									}
								`),
    value: coupon,
    onChange: e => setCoupon(e.target.value)
  })), createElement("div", {
    className: "renderer-core-payment-coupon__apply"
  }, createElement("button", {
    className: classnames({
      loading: isPaying
    }, css`
									border-radius: 8px;
									background: ${generalTheme.buttonsBgColor};
									color: ${generalTheme.buttonsFontColor};
									border: none;
									padding: 10px 20px;
									font-size: 16px;
									cursor: pointer;
									&.loading .renderer-core-arrow-icon {
										display: none;
									}
								`, 'apply-coupon-button'),
    onClick: () => applyCoupon()
  }, createElement("span", {
    id: "button-text"
  }, isPaying ? createElement(Loader, {
    color: generalTheme.buttonsFontColor,
    height: 16,
    width: 16
  }) : createElement(Fragment, null, (_data$payments$labels3 = data.payments.labels?.applyDiscount) !== null && _data$payments$labels3 !== void 0 ? _data$payments$labels3 : 'apply', " "))))), error && createElement("div", {
    className: classnames('renderer-core-payment-coupon__error', css`
							color: #a94442;
							font-size: 14px;
							margin-top: 10px;
						`)
  }, error));
};
export default Coupon;
