// @quillforms/renderer-core/build/components/field-footer/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _data = require("@wordpress/data");
var _editable = _interopRequireDefault(require("./editable"));
var _nonEditable = _interopRequireDefault(require("./non-editable"));
var _fieldRender = require("../field-render");
var _reactLoaderSpinner = require("react-loader-spinner");
var _hooks = require("../../hooks");
var _emotion = require("emotion");
var _parent = _interopRequireDefault(require("./parent"));
/**
 * WordPress Dependencies
 */

/**
 * Internal Dependencies
 */

/**
 * External Dependencies
 */

const BlockFooter = ({
  shakingErr,
  isPending
}) => {
  const {
    id,
    blockName,
    attributes
  } = (0, _fieldRender.__experimentalUseFieldRenderContext)();
  const blockTheme = (0, _hooks.useBlockTheme)(attributes?.themeId);
  if (!blockName) return null;
  const {
    isEditable,
    isParent
  } = (0, _data.useSelect)(select => {
    return {
      isEditable: select('quillForms/blocks').hasBlockSupport(blockName, 'editable'),
      isParent: select('quillForms/blocks').hasBlockSupport(blockName, 'innerBlocks')
    };
  });
  return (0, _react.createElement)("div", {
    className: "renderer-core-field-footer"
  }, isPending ? (0, _react.createElement)("div", {
    className: (0, _emotion.css)`
						margin: 10px;
					`
  }, (0, _react.createElement)(_reactLoaderSpinner.TailSpin, {
    color: blockTheme.answersColor,
    height: 30,
    width: 30
  })) : (0, _react.createElement)(_react.Fragment, null, isParent ? (0, _react.createElement)(_parent.default, null) : (0, _react.createElement)(_react.Fragment, null, !isEditable ? (0, _react.createElement)(_nonEditable.default, null) : (0, _react.createElement)(_editable.default, {
    id: id,
    shakingErr: shakingErr
  }))));
};
var _default = exports.default = BlockFooter;
