// @quillforms/blocklib-statement-block/build/admin/controls.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _adminComponents = require("@quillforms/admin-components");
/**
 * QuillForms Dependencies
 */

/**
 * WordPress Dependencies
 */

const statementControls = props => {
  const {
    attributes,
    setAttributes
  } = props;
  const {
    buttonText,
    quotationMarks
  } = attributes;
  return (0, _react.createElement)(_react.Fragment, null, (0, _react.createElement)(_adminComponents.BaseControl, null, (0, _react.createElement)(_adminComponents.ControlWrapper, {
    orientation: "horizontal"
  }, (0, _react.createElement)(_adminComponents.ControlLabel, {
    label: "Quotation marks"
  }), (0, _react.createElement)(_adminComponents.ToggleControl, {
    checked: quotationMarks,
    onChange: () => setAttributes({
      quotationMarks: !quotationMarks
    })
  }))), (0, _react.createElement)(_adminComponents.BaseControl, null, (0, _react.createElement)(_adminComponents.ControlWrapper, {
    orientation: "vertical"
  }, (0, _react.createElement)(_adminComponents.ControlLabel, {
    label: "Button text"
  }), (0, _react.createElement)(_adminComponents.TextControl, {
    placeholder: "Button Text",
    value: buttonText,
    onChange: val => setAttributes({
      buttonText: val
    })
  }))));
};
var _default = exports.default = statementControls;
