// @quillforms/renderer-core/build/components/form-footer/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _data = require("@wordpress/data");
var _element = require("@wordpress/element");
var _classnames = _interopRequireDefault(require("classnames"));
var _fieldNavigation = _interopRequireDefault(require("../field-navigation"));
var _emotion = require("emotion");
var _progressBar = _interopRequireDefault(require("../progress-bar"));
var _saveBtn = _interopRequireDefault(require("../save-btn"));
var _hooks = require("../../hooks");
var _lodash = require("lodash");
/**
 * WordPress Dependencies
 */

/**
 *
 * External Dependencies
 */

/**
 * Internal Dependencies
 */

const FormFooter = (0, _element.memo)(() => {
  const theme = (0, _hooks.useCurrentTheme)();
  const {
    formObj
  } = (0, _hooks.useFormContext)();
  const messages = (0, _hooks.useMessages)();
  const {
    currentBlockId,
    isWelcomeScreenActive,
    isThankyouScreenActive,
    shouldFooterBeDisplayed,
    correctIncorrectDisplay
  } = (0, _data.useSelect)(select => {
    return {
      currentBlockId: select('quillForms/renderer-core').getCurrentBlockId(),
      isThankyouScreenActive: select('quillForms/renderer-core').isThankyouScreenActive(),
      isWelcomeScreenActive: select('quillForms/renderer-core').isWelcomeScreenActive(),
      shouldFooterBeDisplayed: select('quillForms/renderer-core').shouldFooterBeDisplayed(),
      correctIncorrectDisplay: select('quillForms/renderer-core').getCorrectIncorrectDisplay()
    };
  });
  const {
    isCurrentFieldAnswerCorrect
  } = (0, _data.useSelect)(select => {
    return {
      isCurrentFieldAnswerCorrect: select('quillForms/renderer-core')?.isFieldAnswerCorrect(currentBlockId)
    };
  });
  if (!currentBlockId) return null;
  return (0, _react.createElement)("div", {
    className: (0, _classnames.default)('renderer-components-form-footer', {
      hidden: isWelcomeScreenActive || isThankyouScreenActive
    }, (0, _emotion.css)`
					@media ( min-width: 768px ) {
						background: ${theme.formFooterBgColor.lg} !important;
					}
					@media ( max-width: 767px ) {
						background: ${theme.formFooterBgColor.sm} !important;
					}
				`),
    tabIndex: -1
  }, (0, _react.createElement)("div", {
    className: "renderer-components-form-footer__inner"
  }, (0, _react.createElement)(_saveBtn.default, null), !formObj?.settings?.disableProgressBar && (0, _react.createElement)(_progressBar.default, null), !formObj?.settings?.disableNavigationArrows && (0, _react.createElement)(_fieldNavigation.default, {
    shouldFooterBeDisplayed: shouldFooterBeDisplayed
  }), formObj?.settings?.displayBranding && (0, _react.createElement)("div", {
    className: (0, _classnames.default)(`renderer-core-powered-by-branding-${(0, _lodash.random)(100)}`, (0, _emotion.css)`
								display: block !important;
								visibility: visible !important;

								a {
									display: block !important;
									visibility: visible !important;
								}
							`)
  }, (0, _react.createElement)("a", {
    className: (0, _emotion.css)`
								background: ${theme.buttonsBgColor} !important;
								color: ${theme.buttonsFontColor} !important;
								box-shadow: none !important;
								text-decoration: none !important;
								padding: 5px 8px;
								font-size: 14px;
								outline: none !important;
								z-index: 1;
								border-radius: 5px;
							`,
    href: "https://quillforms.com",
    target: "_blank"
  }, "Powered By Quill Forms")), (0, _react.createElement)("div", {
    className: (0, _classnames.default)(`renderer-core-correct-incorrect-result`, {
      active: correctIncorrectDisplay
    }, (0, _emotion.css)`
							z-index: 999999999;
							color: #fff;
							background: ${isCurrentFieldAnswerCorrect ? '#5bc68a' : '#d93148'};
						`)
  }, isCurrentFieldAnswerCorrect ? messages['label.correct'] : messages['label.incorrect'])));
});
var _default = exports.default = FormFooter;
