// @quillforms/admin-components/build/responsive-control/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ResponsiveControl;
var _react = require("react");
var _element = require("@wordpress/element");
var _i18n = require("@wordpress/i18n");
var _lodash = require("lodash");
var _components = require("@wordpress/components");
var _controlWrapper = _interopRequireDefault(require("../control-wrapper"));
var _controlLabel = _interopRequireDefault(require("../control-label"));
/**
 * External Dependencies
 */

/**
 * Internal Dependencies
 */

function ResponsiveControl({
  label,
  desktopChildren,
  tabletChildren,
  mobileChildren
}) {
  const [deviceType, setDeviceType] = (0, _element.useState)('Desktop');
  const devices = [{
    name: 'Desktop',
    title: "Desktop",
    itemClass: 'qf-desk-tab'
  }, {
    name: 'Mobile',
    key: 'mobile',
    title: 'Mobile',
    itemClass: 'qf-mobile-tab'
  }];
  const output = {};
  output.Mobile = mobileChildren;
  output.Tablet = tabletChildren;
  output.Desktop = desktopChildren;
  return (0, _react.createElement)("div", {
    className: 'admin-components-responsive-control'
  }, (0, _react.createElement)(_controlWrapper.default, {
    orientation: ""
  }, (0, _react.createElement)(_controlLabel.default, {
    label: label
  }), (0, _react.createElement)(_components.ButtonGroup, {
    className: "responsive-options",
    "aria-label": (0, _i18n.__)('Device', 'quillforms')
  }, (0, _lodash.map)(devices, ({
    name,
    key,
    title,
    itemClass
  }) => (0, _react.createElement)(_components.Button, {
    key: key,
    className: `responsive-btn ${itemClass}${name === deviceType ? ' is-active' : ''}`,
    isSmall: true,
    "aria-pressed": deviceType === name,
    onClick: () => setDeviceType(name)
  }, title)))), (0, _react.createElement)("div", {
    className: "measure-control-inner"
  }, output[deviceType] ? output[deviceType] : output.Desktop));
}
