// @quillforms/admin-components/build/combobox-control/use-hidden-fields.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _data = require("@wordpress/data");
var _hiddenFieldIcon = _interopRequireDefault(require("./hidden-field-icon"));
var _lodash = require("lodash");
/**
 * WordPress Dependencies
 */

/**
 * External Dependencies
 */

const useHiddenFields = ({
  section
}) => {
  const {
    hiddenFields
  } = (0, _data.useSelect)(select => {
    var _select$getHiddenFiel;
    return {
      hiddenFields: (_select$getHiddenFiel = select('quillForms/hidden-fields-editor'
      //@ts-ignore
      )?.getHiddenFields()) !== null && _select$getHiddenFiel !== void 0 ? _select$getHiddenFiel : {}
    };
  });
  const fields = [];
  if ((0, _lodash.size)(hiddenFields) > 0) {
    for (const hiddenField of hiddenFields) {
      fields.push({
        type: 'hidden_field',
        value: hiddenField.name,
        label: hiddenField.name,
        iconBox: {
          color: '#aaa',
          icon: _hiddenFieldIcon.default
        },
        section,
        isMergeTag: true
      });
    }
  }
  return fields;
};
var _default = exports.default = useHiddenFields;
