// @quillforms/admin-components/build/combobox-control/use-fields.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _utils = require("@quillforms/utils");
var _data = require("@wordpress/data");
var _lodash = require("lodash");
var _richText = require("../rich-text");
/**
 * Quill Forms Dependencies
 */

/**
 * WordPress Dependencies
 */

/**
 * External Dependencies
 */

const useFields = ({
  section
}) => {
  const {
    blockTypes,
    formBlocks
  } = (0, _data.useSelect)(select => {
    return {
      blockTypes: select('quillForms/blocks').getBlockTypes(),
      // @ts-expect-error
      formBlocks: select('quillForms/block-editor').getBlocks()
    };
  });
  const blocks = [];
  if ((0, _lodash.size)(formBlocks) > 0) {
    (0, _lodash.forEach)(formBlocks, (block, index) => {
      if (blockTypes[block.name]?.supports?.editable) {
        blocks.push({
          ...block,
          order: index
        });
      }
      if (blockTypes[block.name]?.supports?.innerBlocks) {
        if ((0, _lodash.size)(block?.innerBlocks) > 0) {
          (0, _lodash.forEach)(block.innerBlocks, (childBlock, childIndex) => {
            blocks.push({
              ...childBlock,
              order: index + (0, _utils.identAlphabetically)(parseInt(childIndex))
            });
          });
        }
      }
    });
  }
  const fields = [];
  if ((0, _lodash.size)(blocks) > 0) {
    for (const block of blocks) {
      const blockType = blockTypes[block.name];
      fields.push({
        type: 'field',
        value: block.id,
        label: block.attributes?.label ? (0, _richText.getPlainExcerpt)(block.attributes.label) : '',
        iconBox: {
          icon: blockType?.icon,
          color: blockType?.color
        },
        section,
        isMergeTag: true,
        other: {
          name: block.name
        }
      });
    }
  }
  return fields;
};
var _default = exports.default = useFields;
