// @quillforms/admin-components/build/mapping-row/key-control/select/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _selectControl = _interopRequireDefault(require("../../../select-control"));
var _components = require("@wordpress/components");
var _i18n = require("@wordpress/i18n");
var _context = require("../context");
var _option = _interopRequireDefault(require("./option"));
/**
 * QuillForms Dependencies
 */

/**
 * Internal Dependencies
 */

const Select = ({
  sections,
  options
}) => {
  var _SelectOptions$find;
  const {
    value,
    onChange,
    disabled
  } = (0, _context.useMappingKeyControlContext)();

  // @ts-expect-error
  const SelectOptions = [{
    key: 'select',
    name: (0, _react.createElement)(_option.default, {
      label: (0, _i18n.__)('Choose an option', 'quillforms')
    })
  }];
  // add each section with related options.
  for (const section of sections) {
    SelectOptions.push({
      key: section.key,
      name: (0, _react.createElement)("div", {
        onClick: ev => ev.stopPropagation()
      }, section.label),
      className: 'mapping-key-control-select-section'
    });
    options.forEach((option, index) => {
      if (option.section === section.key) {
        SelectOptions.push({
          key: index,
          name: (0, _react.createElement)(_option.default, {
            label: option.label,
            isStarred: option.isStarred,
            hasSection: true
          })
        });
      }
    });
  }
  // add options of undefined section.
  options.forEach((option, index) => {
    if (option.section === undefined) {
      SelectOptions.push({
        key: index,
        name: (0, _react.createElement)(_option.default, {
          label: option.label,
          isStarred: option.isStarred
        })
      });
    }
  });
  const component = (0, _react.createElement)("div", {
    className: "mapping-key-control-select"
  }, (0, _react.createElement)(_selectControl.default, {
    label: "",
    value: (_SelectOptions$find = SelectOptions.find(option => Number.isInteger(option.key) && options[option.key].value === value)) !== null && _SelectOptions$find !== void 0 ? _SelectOptions$find : SelectOptions[0],
    onChange: ({
      selectedItem
    }) => {
      if (selectedItem) {
        if (selectedItem.key === 'select') {
          // clear
          onChange('');
        } else if (Number.isInteger(selectedItem.key)) {
          // update selected value
          const option = options[selectedItem.key];
          onChange(option.value);
        }
      }
    },
    options: SelectOptions
  }));
  return disabled ? (0, _react.createElement)(_components.Disabled, null, component) : component;
};
var _default = exports.default = Select;
