// @quillforms/admin-components/build/rich-text/components/element/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _slateReact = require("slate-react");
var _mergeTag = _interopRequireDefault(require("../merge-tag"));
/**
 * External Dependencies
 */

/**
 * Internal Dependencies
 */

const Element = ({
  attributes,
  children,
  element,
  editor,
  mergeTags
}) => {
  const path = _slateReact.ReactEditor.findPath(editor, element);
  if (element.type === 'mergeTag') {
    return (0, _react.createElement)(_mergeTag.default, {
      element: element,
      mergeTags: mergeTags,
      editor: editor,
      path: path,
      attributes: attributes,
      children: children
    });
  } else if (element.type === 'link') {
    return (0, _react.createElement)("a", {
      ...attributes,
      href: element.url
    }, children);
  }
  return (0, _react.createElement)("p", {
    ...attributes
  }, children);
};
var _default = exports.default = Element;
