// @quillforms/blocks/build/api/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "getBlockType", {
  enumerable: true,
  get: function () {
    return _registration.getBlockType;
  }
});
Object.defineProperty(exports, "registerBlockType", {
  enumerable: true,
  get: function () {
    return _registration.registerBlockType;
  }
});
Object.defineProperty(exports, "sanitizeBlockAttributes", {
  enumerable: true,
  get: function () {
    return _utils.sanitizeBlockAttributes;
  }
});
Object.defineProperty(exports, "sanitizeBlocks", {
  enumerable: true,
  get: function () {
    return _utils.sanitizeBlocks;
  }
});
Object.defineProperty(exports, "setBlockAdminSettings", {
  enumerable: true,
  get: function () {
    return _setBlockAdminSettings.setBlockAdminSettings;
  }
});
Object.defineProperty(exports, "setBlockRendererSettings", {
  enumerable: true,
  get: function () {
    return _setBlockRendererSettings.setBlockRendererSettings;
  }
});
var _setBlockAdminSettings = require("./set-block-admin-settings");
var _setBlockRendererSettings = require("./set-block-renderer-settings");
var _registration = require("./registration");
var _utils = require("./utils");
