// @quillforms/renderer-core/build/store/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.STORE_KEY = exports.SET_THANKYOU_SCREENS = exports.SET_SWIPER_STATE = exports.SET_SUBMISSION_ERR = exports.SET_PAYMENT_DATA = exports.SET_IS_SUBMITTING = exports.SET_IS_REVIEWING = exports.SET_IS_FOCUSED = exports.SET_IS_FIELD_VALID = exports.SET_IS_FIELD_PENDING = exports.SET_IS_FIELD_CORRECT_INCORRECT_SCREEN_DISPLAYED = exports.SET_IS_FIELD_ANSWER_LOCKED = exports.SET_IS_FIELD_ANSWER_CORRECT = exports.SET_IS_FIELD_ANSWERED = exports.SET_IS_CURRENT_BLOCK_SAFE_TO_SWIPE = exports.SET_FOOTER_DISPLAY = exports.SET_FIELD_VALIDATION_ERR = exports.SET_FIELD_PENDING_MSG = exports.SET_FIELD_ANSWER = exports.SET_CORRECT_INCORRECT_DISPLAY = exports.RESET_ANSWERS = exports.INSERT_EMPTY_FIELD_ANSWER = exports.GO_TO_BLOCK = exports.GO_PREV = exports.GO_NEXT = exports.COMPLETE_FORM = void 0;
const SET_SWIPER_STATE = exports.SET_SWIPER_STATE = 'SET_SWIPER_STATE';
const GO_NEXT = exports.GO_NEXT = 'GO_NEXT';
const GO_PREV = exports.GO_PREV = 'GO_PREV';
const GO_TO_BLOCK = exports.GO_TO_BLOCK = 'GO_TO_BLOCK';
const COMPLETE_FORM = exports.COMPLETE_FORM = 'COMPLETE_FORM';
const SET_FIELD_ANSWER = exports.SET_FIELD_ANSWER = 'SET_FIELD_ANSWER';
const INSERT_EMPTY_FIELD_ANSWER = exports.INSERT_EMPTY_FIELD_ANSWER = 'INSERT_EMPTY_FIELD_ANSWER';
const RESET_ANSWERS = exports.RESET_ANSWERS = 'RESET_ANSWERS';
const SET_IS_FIELD_VALID = exports.SET_IS_FIELD_VALID = 'SET_IS_FIELD_VALID';
const SET_IS_FIELD_ANSWERED = exports.SET_IS_FIELD_ANSWERED = 'SET_IS_FIELD_ANSWERED';
const SET_IS_FIELD_PENDING = exports.SET_IS_FIELD_PENDING = 'SET_IS_FIELD_PENDING';
const SET_FIELD_PENDING_MSG = exports.SET_FIELD_PENDING_MSG = 'SET_FIELD_PENDING_MSG';
const SET_FIELD_VALIDATION_ERR = exports.SET_FIELD_VALIDATION_ERR = 'SET_FIELD_VALIDATION_ERR';
const SET_IS_FIELD_ANSWER_CORRECT = exports.SET_IS_FIELD_ANSWER_CORRECT = 'SET_IS_FIELD_ANSWER_CORRECT';
const SET_IS_FIELD_ANSWER_LOCKED = exports.SET_IS_FIELD_ANSWER_LOCKED = 'SET_IS_FIELD_ANSWER_LOCKED';
const SET_IS_CURRENT_BLOCK_SAFE_TO_SWIPE = exports.SET_IS_CURRENT_BLOCK_SAFE_TO_SWIPE = 'SET_IS_CURRENT_BLOCK_SAFE_TO_SWIPE';
const SET_THANKYOU_SCREENS = exports.SET_THANKYOU_SCREENS = 'SET_THANKYOU_SCREENS';
const SET_IS_FIELD_CORRECT_INCORRECT_SCREEN_DISPLAYED = exports.SET_IS_FIELD_CORRECT_INCORRECT_SCREEN_DISPLAYED = 'SET_IS_FIELD_CORRECT_INCORRECT_SCREEN_DISPLAYED';
const SET_CORRECT_INCORRECT_DISPLAY = exports.SET_CORRECT_INCORRECT_DISPLAY = 'SET_CORRECT_INCORRECT_DISPLAY';
const SET_IS_REVIEWING = exports.SET_IS_REVIEWING = 'SET_IS_REVIEWING';
const SET_SUBMISSION_ERR = exports.SET_SUBMISSION_ERR = 'SET_SUBMISSION_ERR';
const SET_IS_SUBMITTING = exports.SET_IS_SUBMITTING = 'SET_IS_SUBMITTING';
const SET_PAYMENT_DATA = exports.SET_PAYMENT_DATA = 'SET_PAYMENT_DATA';
const SET_IS_FOCUSED = exports.SET_IS_FOCUSED = 'SET_IS_FOCUSED';
const SET_FOOTER_DISPLAY = exports.SET_FOOTER_DISPLAY = 'SET_FOOTER_DISPLAY';
const STORE_KEY = exports.STORE_KEY = 'quillForms/renderer-core';
